src/main/java/uk/gov/companieshouse/filinghistory/consumer/mapper/ChildNodeMapperFactory.java

```diff
@@ -8,17 +8,20 @@ public class ChildNodeMapperFactory {
 
     private final AnnotationNodeMapper annotationNodeMapper;
     private final AssociatedFilingNodeMapper associatedFilingNodeMapper;
+    private final EmbeddedChildNodeMapper embeddedChildNodeMapper;
 
     public ChildNodeMapperFactory(AnnotationNodeMapper annotationNodeMapper,
-                                  AssociatedFilingNodeMapper associatedFilingNodeMapper) {
+                                  AssociatedFilingNodeMapper associatedFilingNodeMapper, EmbeddedChildNodeMapper embeddedChildNodeMapper) {
         this.annotationNodeMapper = annotationNodeMapper;
         this.associatedFilingNodeMapper = associatedFilingNodeMapper;
+        this.embeddedChildNodeMapper = embeddedChildNodeMapper;
     }
 
     public ChildNodeMapper getChildMapper(TransactionKindEnum kind) {
         return switch (kind) {
             case ANNOTATION -> annotationNodeMapper;
             case ASSOCIATED_FILING -> associatedFilingNodeMapper;
+            case TOP_LEVEL -> embeddedChildNodeMapper; // TODO: Check if TOP_LEVEL makes sense here?
```

Comment:
Makes sense to me, we only get here if its a parent (top level transaction) with a child embedded in it. If other edge cases arise in the future we can change it.